astropy/table/tests/test_column.py

```diff
@@ -773,6 +774,8 @@ def test_col_unicode_sandwich_unicode():
     assert ok.dtype.char == '?'
     assert np.all(ok)
 
+    # The following emits a FutureWarning in numpy >=1.24, which is OK,
+    # so filtered out above.
```

Comment:
If only this line is expected to emit that warning, it is safer to do a `warnings.filterwarnings` but only for this line using a `warnings.catch_warnings` context manager.

Reply:
OK, makes sense. Done.